### DISCUSSION TITLE
Repeated Imports in Client Components: Bundle Impact?

### DISCUSSION THREAD

**Discussion by llaalgol** [vercel/next.js#63752] [Help] [answered]
### Summary

If I import the same large library into multiple 'use client' components, will it increase my app's bundle size on the client due to redundant imports, or will the compiler optimize this automatically?

**Comment by icyJoseph** [comment, marked as answer]
Uff, the naive answer is that these are bundled so that they are shareable. For example, if I use `swr` in two page client components, then I'd see this bundle:

<img width="423" alt="Screenshot 2024-03-27 at 09 13 41" src="https://github.com/vercel/next.js/assets/21013447/d97ebb28-9578-4aa8-a4e9-e52b3fef72fb">

Which I can see further in `.next/app-build-manifest.json`:

```json
   "/about/page": [
      "static/chunks/webpack-0dbc3eab889c5ca4.js",
      "static/chunks/fd9d1056-60904e6732b40dac.js",
      "static/chunks/69-fc4eb2bab88217a3.js",
      "static/chunks/main-app-00c9474ce2170511.js",
      "static/chunks/16-6b80c9456318f4fe.js",
      "static/chunks/app/about/page-15c91a8849525441.js"
    ],
    "/page": [
      "static/chunks/webpack-0dbc3eab889c5ca4.js",
      "static/chunks/fd9d1056-60904e6732b40dac.js",
      "static/chunks/69-fc4eb2bab88217a3.js",
      "static/chunks/main-app-00c9474ce2170511.js",
      "static/chunks/16-6b80c9456318f4fe.js",
      "static/chunks/app/page-8ced7d09589d7c64.js"
    ]
  }

```

So we see that `/about/page` and `/page`, are referencing the same module.

A good practice is to examine your bundle regularly, you can use: https://www.npmjs.com/package/@next/bundle-analyzer, which is what I used to generate the image above.

**Reply by llaalgol** [reply]
Your assistance is greatly appreciated; it's truly helpful
